expertise/service/utils.py

```diff
@@ -158,3 +159,234 @@ def get_user_id(openreview_client):
     """
     user = openreview_client.user
     return user.get('user', {}).get('id') if user else None
+
+def _get_required_field(req, superkey, key):
+    try:
+        field = req.pop(key)
+    except KeyError:
+        raise openreview.OpenReviewException(f"Bad request: required field missing in {superkey}: {key}")
+    return field
+
+class APIRequest(object):
+    """
+    Validates and load objects and fields from POST requests
+    """
+    def __init__(self,
+            name = None,
+            user_id = None,
+            job_id = None,
+            token = None,
+            baseurl = None,
+            baseurl_v2 = None):
+            
+        self.name = name
+        self.user_id = user_id
+        self.job_id = job_id
+        self.token = token
+        self.baseurl = baseurl
+        self.baseurl_v2 = baseurl_v2
+        self.entityA = {}
+        self.entityB = {}
+        self.model = {}
+
+    def from_json(self, request = {}):
+        root_key = 'request'
+        def _get_field_from_request(field):
+            return _get_required_field(request, root_key, field)
+
+        def _load_entity_a(entity):
+            self._load_entity('entityA', entity, self.entityA)
+
+        def _load_entity_b(entity):
+            self._load_entity('entityB', entity, self.entityB)
+
+        def _camel_to_snake(camel_str):
+            camel_str = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', camel_str)
+            return re.sub('([a-z0-9])([A-Z])', r'\1_\2', camel_str).lower()
+        
+        # Get the name of the job
+        self.name = _get_field_from_request('name')
+
+        # These are automatically provided by the server
+        self.user_id = _get_field_from_request('user_id')
+        self.job_id = _get_field_from_request('job_id')
```

Comment:
why the request has a job_id?

Reply:
I can probably improve this. Some of these metadata fields get set `routes.py`, some get set in `start_expertise()` and `_prepare_config()` in `expertise.py` but I should move all of these fields into a single spot